doctest/extensions/doctest_mpi.h

```diff
@@ -2,80 +2,104 @@
 
 #ifdef DOCTEST_CONFIG_IMPLEMENT
 
-#include "doctest/doctest.h"
+#include "doctest/extensions/mpi_sub_comm.h"
 #include "mpi_reporter.h"
-
-#else
-
-#include "mpi.h"
-#include <numeric>
-#include <vector>
-#include "doctest/doctest.h"
-#include <cassert>
-#include <string>
+#include <unordered_map>
 
 namespace doctest {
 
-inline
-int mpi_world_nb_procs() {
-  int n;
-  MPI_Comm_size(MPI_COMM_WORLD, &n);
-  return n;
+// Each time a MPI_TEST_CASE is executed on N procs,
+// we need a sub-communicator of N procs to execute it.
+// It is then registered here and can be re-used
+// by other tests that requires a sub-comm of the same size
+std::unordered_map<int,mpi_sub_comm> sub_comms_by_size;
+
+// Record if at least one MPI_TEST_CASE was registered "skipped"
+// because there is not enought procs to execute it
+int nb_test_cases_skipped_insufficient_procs = 0;
+
+// Can be safely called before MPI_Init()
```

Comment:
Shouldn't this be `doctest::mpi_init_thread` instead of `MPI_Init`? (same as a few lines below)

Reply:
I think we could put `MPI_Init_thread ` here since this is the one used by `doctest::mpi_init_thread`, but both `MPI_Init` and `MPI_Init_thread `  essentially do the same thing and `MPI_Init `is better known, which is why I put this name here